File: Alok/Java/src/LcmHcf.java
import java.util.Scanner;

public class LcmHcf {
    public static int hcf(int a, int b){
        //another way
        // if(b % a == 0){
        //     return a;
        // }

        // return hcf(b % a, a);
        while(a != 0){
            int temp = a;
            a = b % a;
            b = temp;
        }
        return b;
    }
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n1 = in.nextInt();
        int n2 = in.nextInt();
        int gcd = hcf(n1, n2);
        System.out.println("HCF is: " + gcd);
        System.out.println("LCM is: " + (n1 * n2) / gcd);

    }
    
}

File: bunty/lcm_hcf.cpp
#include<bits/stdc++.h>
using namespace std;


int gcd(int a, int b){
 
    while(a!=0){
        int temp=a;
        a=b%a;
        b=temp;
    }
    return b;
}

int lcm(int a, int b){
    return a*b/(gcd(a,b));
}

int megatron_prime(int n){

    bool sahi=true;

    for(int i=2;i*i<=n;i++){
        if(n%i==0){
            sahi=false;
            break;
        }
    }
    return sahi;
}

int main(){
    

    cout<<lcm(18,4)<<endl;
    cout<<gcd(18,4)<<endl;
    cout<<megatron_prime(7)<<endl;

    return 0;
}